Dockerfile

```diff
@@ -0,0 +1,46 @@
+FROM mcr.microsoft.com/powershell:alpine-3.12
+
+ARG CLI_VERSION=latest
+
+LABEL name="m365pnp/cli-microsoft365:${CLI_VERSION}" \
+  description="Manage Microsoft 365 and SharePoint Framework projects on any platform" \
+  homepage="https://pnp.github.io/cli-microsoft365" \
+  maintainers="Waldek Mastykarz <waldek@mastykarz.nl>, \
+  Velin Georgiev <velin.georgiev@gmail.com>, \
+  Garry Trinder <garry.trinder@live.com>, \
+  Albert-Jan Schot <appie@digiwijs.nl>, \
+  Rabia Williams <rabiawilliams@gmail.com>, \
+  Mark Powney <mpowney@icloud.com>, \
+  Patrick Lamber <patrick@nubo.eu>" \
+  com.azure.dev.pipelines.agent.handler.node.path="/usr/bin/node"
+
+RUN apk add --no-cache \
+  curl \
+  sudo \
+  bash \
+  shadow \
+  bash-completion \
+  nodejs=12.18.4-r0 \
```

Comment:
Let's update to Node@14 to keep the runtime version in sync with our CI pipe

Reply:
@waldekmastykarz Ok so this was more complicated than it first seemed... 

As we are using alpine linux as the base image, this forces us down the route of using the `apk` package manager and it seems to work in a way that packages are tied to the version of the alpine image used.

For example, the highest version of `nodejs` available on the alpine 3.12 main branch is v12, to get node v14 you can use the `nodejs-current` package but is a community package and there is no corresponding npm package for this version, and we need `npm` installed to perform the installation of packages needed by the CLI to run, so that isn't going to work.

See packages available

<table>
<tr>
	<th>Node
	<th>Name
	<th>Branch
	<th>Repo
	<th>URL
<tr>
	<td>v12
	<td>nodejs 
	<td>3.12
	<td>main 
	<td><a href="https://pkgs.alpinelinux.org/packages?name=nodejs&branch=v3.12&repo=main">https://pkgs.alpinelinux.org/packages?name=nodejs&branch=v3.12&repo=main</a>
<tr>
	<td>v14
	<td>nodejs-current 
	<td>3.12
	<td>community 
	<td><a href="https://pkgs.alpinelinux.org/packages?name=nodejs&branch=v3.12&repo=main">https://pkgs.alpinelinux.org/packages?name=nodejs&branch=v3.12&repo=main</a>
<tr>
	<td>v14
	<td>nodejs 
	<td>edge
	<td>main 
	<td><a href="https://pkgs.alpinelinux.org/packages?name=nodejs&branch=v3.12&repo=main">https://pkgs.alpinelinux.org/packages?name=nodejs&branch=v3.12&repo=main</a>
</table>

The only way to get `nodejs` v14 and the corresponding `npm` version, we have to target the main repository on the edge branch, the edge branch being the development branch of alpine. 

I've made the change and confirmed it works, but not sure if its worth waiting till the v14 package drops into the 3.12 main repository rather than targeting edge.

What do you suggest? Use the edge repo or upgrade later?

Reply:
@waldekmastykarz _bump_ 😄

Reply:
In that case, I'd settle on using an official package even if it means that for the time being we're on Node@12. Do you have any more information about when Node@14 is expected to land?

Reply:
I agree on using official packages, I'll update.

I'm not sure but I can look into finding out a time frame.

Reply:
Dropped back to `node@12` in latest push.